### PR TITLE
Edit project

### LINKED ISSUE #481: Edit project
We probably want to edit a project. This is not possible right now. This issue tracks the implementation of that feature.

### PR DESCRIPTION
**Pull request by jitsedesmet**
This PR closes #481.

### CODE REVIEW

File: frontend/src/components/project/projectForm/createGoalSubForm.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import useTranslation from "next-translate/useTranslation";
+import { Col, Row } from "react-bootstrap";
+import { capitalize } from "../../../utility/stringUtil";
+import Image from "next/image";
+import { ChangeEvent, useState } from "react";
+
+/**
+ * Properties needed by [CreateGoalsSubFormProps].
```
Comment (KasperDemeyere) [inline frontend/src/components/project/projectForm/createGoalSubForm.tsx:8]:
```suggestion
 * Properties needed by [CreateGoalsSubForm].
```

File: frontend/src/components/project/projectForm/projectForm.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+}
+
+type AlteredSkillMapper = { [projectSkillUrl: string]: ProjectSkill };
+
+/**
+ * A form that allows you to create or edit a project.
+ * Includes the creation of projectSkills.
+ * Allows you to link coaches to the project.
+ * @param project the project that needs to be edited. Undefined when in project creation mode.
+ * @constructor
```
Comment (KasperDemeyere) [inline frontend/src/components/project/projectForm/projectForm.tsx:54]:
```suggestion
```

File: frontend/src/components/project/projectForm/projectForm.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+              partnerName: "",
+              partnerWebsite: "",
+          };
+
+    return (
+        <div className={styles.create_project_box} data-testid="create-project-form w-100">
+            <Formik initialValues={initialValues} enableReinitialize={true} onSubmit={handleSubmit}>
+                <Form
+                    onKeyPress={(e) => {
+                        e.which === 13 && e.preventDefault();
```
Comment (KasperDemeyere) [inline frontend/src/components/project/projectForm/projectForm.tsx:166]: Maybe add a small comment explaining what this does.

File: frontend/src/pages/projects/[id]/edit.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import styles from "../../../styles/Home.module.css";
+import NavBar from "../../../components/util/navBar";
+import { ProjectForm } from "../../../components/project/projectForm/projectForm";
+import { projectFormSubmitHandler } from "../../../handlers/projectFormSubmitHandler";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import apiPaths from "../../../properties/apiPaths";
+import { getProjectOnUrl } from "../../../api/calls/projectCalls";
+import { emptyProject, IProject } from "../../../api/entities/ProjectEntity";
+
```
Comment (KasperDemeyere) [inline frontend/src/pages/projects/[id]/edit.tsx:13]: Add some docs.

File: frontend/src/pages/projects/[id]/index.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { NextPage } from "next";
-import { ProjectInfo } from "../../components/project/projectInfo";
-import NavBar from "../../components/util/navBar";
-import { ProjectList } from "../../components/project/projectList";
-import styles from "../../styles/pageGrids.module.css";
+import styles from "../../../styles/pageGrids.module.css";
+import NavBar from "../../../components/util/navBar";
+import { ProjectList } from "../../../components/project/projectList";
+import { ProjectInfo } from "../../../components/project/projectInfo";
```
Comment (KasperDemeyere) [inline frontend/src/pages/projects/[id]/index.tsx:6]: Add some docs.

File: frontend/src/pages/projects/create.tsx [hunk 6]
```diff
@@ -1,17 +1,14 @@
 import type { NextPage } from "next";
-import useTranslation from "next-translate/useTranslation";
-import { CreateProjectForm } from "../../components/project/createProjectForm";
-import { createProjectSubmitHandler } from "../../handlers/createProjectSubmitHandler";
 import NavBar from "../../components/util/navBar";
 import styles from "../../styles/projects/createProject.module.css";
+import { ProjectForm } from "../../components/project/projectForm/projectForm";
```
Comment (KasperDemeyere) [inline frontend/src/pages/projects/create.tsx:5]: Add some docs.

**Review by KasperDemeyere — CHANGES_REQUESTED** [review]
Some docs need to be changed/added.

**Review by ruvmello — DISMISSED** [review]
No real complaints. I'm not gonna be nitpicking the last days before the deadline. I see you wrote a lot of tests too, so very good work! Didn't notice very weird things so this can be approved.

**Review by KasperDemeyere — APPROVED** [review]

**Pull request merged**